modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java

```diff
@@ -869,6 +892,84 @@ private void _testPatchAccountWithContactInformation() throws Exception {
 			patchAccount.getAccountContactInformation());
 	}
 
+	private void _testPatchAccountWithMoreExternalReferenceCodes()
+		throws Exception {
+
+		Account postAccount = testPatchAccount_addAccount();
+
+		Account parentAccount = randomAccount();
+		Account randomPatchAccount = randomPatchAccount();
+
+		Account postParentAccount = testPostAccount_addAccount(parentAccount);
+
+		Organization organization1 = OrganizationTestUtil.addOrganization();
+		Organization organization2 = OrganizationTestUtil.addOrganization();
+		Address billingAddress = OrganizationTestUtil.addAddress(organization1);
+		Address shippingAddress = OrganizationTestUtil.addAddress(
+			organization2);
+
+		FileEntry fileEntry = _addFileEntry();
+
+		randomPatchAccount.setDefaultBillingAddressExternalReferenceCode(
+			billingAddress.getExternalReferenceCode());
+		randomPatchAccount.setDefaultBillingAddressId(0L);
+
+		randomPatchAccount.setDefaultShippingAddressExternalReferenceCode(
+			shippingAddress.getExternalReferenceCode());
+		randomPatchAccount.setDefaultShippingAddressId(0L);
+
+		randomPatchAccount.setLogoExternalReferenceCode(
+			fileEntry.getExternalReferenceCode());
+		randomPatchAccount.setLogoId(0L);
+
+		randomPatchAccount.setOrganizationExternalReferenceCodes(
+			new String[] {
+				organization1.getExternalReferenceCode(),
+				organization2.getExternalReferenceCode()
+			});
+		randomPatchAccount.setOrganizationIds(new Long[0]);
+
+		randomPatchAccount.setParentAccountExternalReferenceCode(
+			postParentAccount.getExternalReferenceCode());
+		randomPatchAccount.setParentAccountId(0L);
+
+		Account patchAccount = accountResource.patchAccount(
+			postAccount.getId(), randomPatchAccount);
+
+		randomPatchAccount.setDefaultBillingAddressId(
+			billingAddress.getAddressId());
+		randomPatchAccount.setDefaultShippingAddressId(
+			shippingAddress.getAddressId());
+		randomPatchAccount.setLogoId(fileEntry.getFileEntryId());
+		randomPatchAccount.setOrganizationIds(
+			new Long[] {
+				organization1.getOrganizationId(),
+				organization2.getOrganizationId()
+			});
+		randomPatchAccount.setParentAccountId(postParentAccount.getId());
+
+		Arrays.sort(randomPatchAccount.getOrganizationIds());
+		Arrays.sort(patchAccount.getOrganizationIds());
+
+		Assert.assertEquals(
+			randomPatchAccount.getDefaultBillingAddressId(),
+			patchAccount.getDefaultBillingAddressId());
+		Assert.assertEquals(
+			randomPatchAccount.getDefaultShippingAddressId(),
+			patchAccount.getDefaultShippingAddressId());
+		//		Assert.assertEquals(
+		//			randomAccount.getLogoId(),
+		//			postAccount.getLogoId());
+		Assert.assertArrayEquals(
+			randomPatchAccount.getOrganizationIds(),
+			patchAccount.getOrganizationIds());
+		Assert.assertEquals(
+			randomPatchAccount.getParentAccountId(),
+			patchAccount.getParentAccountId());
+
+		assertValid(patchAccount);
```

Comment:
remove this (it should be already validate by the "default patch test" or move before all the Assert.assertEquals